### ISSUE TITLE
Support Powerline enabled fonts

### ISSUE THREAD

**Issue by sedwards2009** [sedwards2009/extraterm#77]
The first problem is that under Linux/KDE, powerline fonts like Fira are not selectable from the Settings tab.

Second problem is that the shell integration conflicts with that from Powerline itself.

**Comment by Gromo13** [comment]
Linux/Gnome (Ubuntu 17.10) has same problem.

**Comment by sedwards2009** [comment]
In the next release it will be possible to select more fonts.

**Comment by Gromo13** [comment]
Yeah, there is one font with powerline symbols, but it still looks awful.

**Comment by sedwards2009** [comment]
@Gromo13  can you attach a screenshot?

**Comment by Gromo13** [comment]
[https://imgur.com/JORiogJ](example of extraterm with Powerline font)

I tried almost every font included in  extraterm and only this one is supporting powerline symbols.

**Comment by sedwards2009** [comment]
I'm just guessing from the name of the font that it only contains those few powerline symbols and nothing else. This would explain why the letters are using some ugly default font.

Try installing a font from https://github.com/powerline/fonts which has been patched. Those monospaced patched fonts should work better.

**Comment by deathtrip** [comment]
On Arch Linux I have powerline-fonts installed, but the only font that works in Extraterm with powerline symbols is Hack. In other terminal emulators i get all supported fonts. Not sure if anyone can confirm the issue.

**Comment by sedwards2009** [comment]
@btxtiger You mean those standard system fonts just are not in the list?

**Comment by btxtiger** [comment]
Yes, it just shows a few fonts, mainly non-mono like Arial, Comic Sans, Tahoma, Verdana

**Comment by sedwards2009** [comment]
@btxtiger The commit I just put on this issue should fix the problem. It will be in the next release.

**Comment by kdziadowiec** [comment]
I am also using Hack font (on Mac), but for me the problem is that the symbols are all weirdly cut off. I have no idea why is it this way, but exactly the same font displays well in iTerm2.

![image](https://user-images.githubusercontent.com/45563326/52718315-732cc780-2fa3-11e9-84c5-b6484c0e6005.png)
![image](https://user-images.githubusercontent.com/45563326/52718380-98213a80-2fa3-11e9-8fd7-e0d67ae282d3.png)

**Comment by sedwards2009** [comment]
@kdziadowiec I have plans to change how the font and text rendering is done. There I will tackle these font compatibility problems, rendering glitches and hopefully improved speed.

**Comment by Mario253** [comment]
I'm getting the cut off symbols as well, with the heart for battery life being hidden behind the bar displaying battery life.

![selection_001](https://user-images.githubusercontent.com/32368758/52970015-038a5400-3378-11e9-9825-2457b14fc03c.png)

**Comment by ksawerykarwacki** [comment]
@sedwards2009 do you have any timeline for this issue? I would love to switch from conemu to extraterm, but i use a lot of custom symbols from nerd fonts and have look&feel set to use those powerline arrows. Maybe it is not a big isssue for most of the people but those gaps on the triangles triggers my OCD every time I look at the terminal ;)

**Comment by sedwards2009** [comment]
@ksawerykarwacki There are a bunch of related glyph rendering issues at the moment which also drive me nuts. Custom fonts is part of that. The fix will be canvas based rendering instead of the current DOM based one. This will give me a lot more control over how to the drawing is done and also it should be faster too.

I've updated the Development Roadmap: https://github.com/sedwards2009/extraterm/issues/30

**Comment by krzysztof-callstats** [comment]
Sorry, but where exactly is it in the roadmap? I can't find it unless it is under some different name 🤔

**Comment by sedwards2009** [comment]
@krzysztof-callstats  It may have fallen off the map for the time being. I'm not quite sure how best to hang this. Either allow multiple fonts to be "stacked" and used, or to do it via a plugin system. The underlying code to do this mostly exists. The graphics drawing characters are custom drawn and not from a font.

The good news is that using a custom font with the extra glyphs built in might work ok. I heard that Chinese characters are now rendering better than before the summer.
